File: fullstack/ruby/price_calculator/lib/cart/models/cart_item.rb
class CartItem < Struct.new(:product_type, :options, :artist_markup, :quantity, :base_price)
  def calculate_price
    # raise NotImplementedError
  end

  def self.from_hash(hash)
    ProductBasePrice.new(
      hash.fetch('product-type'),
      hash.fetch('options'),
      hash.fetch('artist-markup'),
      hash.fetch('quantity'),
      nil
    )
  end
end

File: fullstack/js/price-calculator/app/models/cart-item.js
export class CartItem {

    productType = "";

    options = {};

    /**
     * The artist markup in percentile. E.g. 20 would be 20% in markup.
     */
    artistMarkup = 0;

    /**
     * The quantity of this item.
     */
    quantity = 0;

    /**
     * The base price object of this item
     */
    basePrice;

    constructor(productType, options, artistMarkup, quantity, basePrice) {
        this.productType = productType;
        this.options = options;
        this.artistMarkup = artistMarkup;
        this.quantity = quantity;
        this.basePrice = basePrice;
    }
}
